app/javascript/packs/diagnosis/addContactModal.vue.erb

```diff
@@ -0,0 +1,175 @@
+                            </div>
+                            <p>
+                                <%= I18n.t('diagnosis.show.add_contact_modal.error_message') %>
+                            </p>
+                        </div>
+                        <div class="required field" v-bind:class="{ error: !nameCompleted && errorShowing }">
+                            <label><%= I18n.t('diagnosis.show.add_contact_modal.full_name') %></label>
+                            <input v-model="name" type="text">
+                        </div>
+                        <div class="required field" v-bind:class="{ error: !jobCompleted && errorShowing }">
+                            <label><%= I18n.t('diagnosis.show.add_contact_modal.role') %></label>
+                            <input v-model="job" type="text" required="required">
+                        </div>
+                        <h3 class="ui header">
+                            <%= I18n.t('diagnosis.show.add_contact_modal.contact_detail') %>
+                            <div class="sub header "><%= I18n.t('diagnosis.show.add_contact_modal.add_one_contact_mode_below') %></div>
+                        </h3>
+                        <div class="field" v-bind:class="{ error: !contactCompleted && errorShowing }">
+                            <label><%= I18n.t('diagnosis.show.add_contact_modal.email_address') %></label>
+                            <input v-model="email" type="text">
+                        </div>
+                        <div class="field"  v-bind:class="{ error: !contactCompleted && errorShowing }">
+                            <label><%= I18n.t('diagnosis.show.add_contact_modal.phone_number') %></label>
+                            <input v-model="phoneNumber" type="text">
+                        </div>
+                    </div>
+                </div>
+            </div>
+            <div class="actions">
+                <div class="ui right labeled icon button" v-on:click="closeButtonClicked"
+                v-bind:class="{ disabled: isSaving}">
+                    <%= I18n.t('cancel') %>
+                    <i class="remove icon"></i>
+                </div>
+                <button class="ui positive right labeled icon button" v-on:click="saveButtonClicked"
+                 v-bind:class="{ disabled: isSaving, loading: isSaving, button: isSaving}">
+                    <%= I18n.t('save') %>
+                    <i class="checkmark icon"></i>
+                </button>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+    import RequestService from './requestService'
+
+    import { mapMutations, mapActions } from 'vuex'
+    import * as types from './store/mutationTypes'
+
+    export default {
+        name: 'add-contact-modal',
+        props: [
+            'show',
+            'visitId'
+            ],
+        data: function () {
+            return {
```

Comment:
Why a function returning a hash instead of just a hash ?
Ex : 
```
data: {
            name: '',
            job: '',
            email: '',
            phoneNumber: '',
            errorShowing: false,
            showErrorMessage: false,
            isSaving: false
        },
```